### ISSUE TITLE
[3.1.0] Does not compile out-of-the-box due to "enforcer" dependency errors

### ISSUE THREAD

**Issue by bicschneider** [Customatics/leaptest-plugin#22]
```
[INFO] Scanning for projects...
[WARNING] The POM for org.jenkins-ci.tools:maven-hpi-plugin:jar:2.1 is missing, no dependency information available
[WARNING] Failed to build parent project for org.jenkins-ci.plugins:leaptest-plugin:hpi:3.1.0
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Leapwork Integration 3.1.0
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-hpi-plugin:2.1:validate (default-validate) @ leaptest-plugin ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:display-info (display-info) @ leaptest-plugin ---
[INFO] Maven Version: 3.3.9
[INFO] JDK Version: 1.8.0_191 normalized as: 1.8.0-191
[INFO] OS Info: Arch: amd64 Family: windows Name: windows 10 Version: 10.0
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (display-info) @ leaptest-plugin ---
[INFO] Ignoring requireUpperBoundDeps in com.google.guava:guava
[WARNING] Rule 5: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for org.apache.ant:ant:1.8.4 paths to dependency are:
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-core:2.60.3
    +-org.apache.ant:ant:1.8.4
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.ant:ant:1.9.2
, 
Require upper bound dependencies error for org.eclipse.jetty:jetty-util:9.2.12.v20150709 paths to dependency are:
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.main:jenkins-test-harness-htmlunit:2.18-1
      +-org.eclipse.jetty:jetty-util:9.2.12.v20150709
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.eclipse.jetty:jetty-webapp:9.2.15.v20160210
      +-org.eclipse.jetty:jetty-xml:9.2.15.v20160210
        +-org.eclipse.jetty:jetty-util:9.2.15.v20160210
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.main:jenkins-test-harness-htmlunit:2.18-1
      +-org.eclipse.jetty.websocket:websocket-client:9.2.12.v20150709
        +-org.eclipse.jetty:jetty-util:9.2.12.v20150709
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.main:jenkins-test-harness-htmlunit:2.18-1
      +-org.eclipse.jetty:jetty-io:9.2.12.v20150709
        +-org.eclipse.jetty:jetty-util:9.2.12.v20150709
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.main:jenkins-test-harness-htmlunit:2.18-1
      +-org.eclipse.jetty.websocket:websocket-common:9.2.12.v20150709
        +-org.eclipse.jetty:jetty-util:9.2.12.v20150709
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.eclipse.jetty:jetty-security:9.2.15.v20160210
      +-org.eclipse.jetty:jetty-server:9.2.15.v20160210
        +-org.eclipse.jetty:jetty-http:9.2.15.v20160210
          +-org.eclipse.jetty:jetty-util:9.2.15.v20160210
, 
Require upper bound dependencies error for org.eclipse.jetty:jetty-io:9.2.12.v20150709 paths to dependency are:
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.main:jenkins-test-harness-htmlunit:2.18-1
      +-org.eclipse.jetty:jetty-io:9.2.12.v20150709
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.eclipse.jetty:jetty-security:9.2.15.v20160210
      +-org.eclipse.jetty:jetty-server:9.2.15.v20160210
        +-org.eclipse.jetty:jetty-io:9.2.15.v20160210
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.main:jenkins-test-harness-htmlunit:2.18-1
      +-org.eclipse.jetty.websocket:websocket-client:9.2.12.v20150709
        +-org.eclipse.jetty:jetty-io:9.2.12.v20150709
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.main:jenkins-test-harness-htmlunit:2.18-1
      +-org.eclipse.jetty.websocket:websocket-common:9.2.12.v20150709
        +-org.eclipse.jetty:jetty-io:9.2.12.v20150709
, 
Require upper bound dependencies error for org.codehaus.plexus:plexus-utils:2.1 paths to dependency are:
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.eclipse.sisu:org.eclipse.sisu.plexus:0.1.0
      +-org.codehaus.plexus:plexus-utils:2.1
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.maven:maven-core:3.1.0
        +-org.codehaus.plexus:plexus-utils:3.0.10
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.maven:maven-artifact:3.1.0
        +-org.codehaus.plexus:plexus-utils:3.0.10
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.maven:maven-compat:3.1.0
        +-org.codehaus.plexus:plexus-utils:3.0.10
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.maven:maven-embedder:3.1.0
        +-org.codehaus.plexus:plexus-utils:3.0.10
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.maven:maven-aether-provider:3.1.0
        +-org.codehaus.plexus:plexus-utils:3.0.10
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.maven.wagon:wagon-provider-api:2.4
        +-org.codehaus.plexus:plexus-utils:3.0.8
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.maven:maven-core:3.1.0
        +-org.apache.maven:maven-model:3.1.0
          +-org.codehaus.plexus:plexus-utils:3.0.10
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.maven:maven-core:3.1.0
        +-org.apache.maven:maven-settings:3.1.0
          +-org.codehaus.plexus:plexus-utils:3.0.10
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.maven:maven-core:3.1.0
        +-org.apache.maven:maven-settings-builder:3.1.0
          +-org.codehaus.plexus:plexus-utils:3.0.10
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.maven:maven-core:3.1.0
        +-org.apache.maven:maven-repository-metadata:3.1.0
          +-org.codehaus.plexus:plexus-utils:3.0.10
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.maven:maven-core:3.1.0
        +-org.apache.maven:maven-model-builder:3.1.0
          +-org.codehaus.plexus:plexus-utils:3.0.10
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.maven:maven-core:3.1.0
        +-org.sonatype.plexus:plexus-sec-dispatcher:1.3
          +-org.codehaus.plexus:plexus-utils:1.5.5
, 
Require upper bound dependencies error for org.codehaus.plexus:plexus-classworlds:2.4 paths to dependency are:
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.eclipse.sisu:org.eclipse.sisu.plexus:0.1.0
      +-org.codehaus.plexus:plexus-classworlds:2.4
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.maven:maven-core:3.1.0
        +-org.codehaus.plexus:plexus-classworlds:2.4.2
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.maven:maven-embedder:3.1.0
        +-org.codehaus.plexus:plexus-classworlds:2.4.2
]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 14.877 s
[INFO] Finished at: 2019-01-22T13:33:28+01:00
[INFO] Final Memory: 48M/620M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M1:enforce (display-info) on project leaptest-plugin: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

Process finished with exit code 1
```

**Comment by romanchetto** [comment]
Hello @bicschneider 
I've spent 2 days to find the reason and fix it, but not succeded.
To build I use command `clean package -Denforcer.skip=true `
If you have an idea how to fix it, please let me know.

**Comment by bicschneider** [comment]
Just use my pom.xml file from my fork. It should work as I recall .. Or set the upper bound dependency explicitely in the pom file
```
Require upper bound dependencies error for org.apache.ant:ant:1.8.4 paths to dependency are:
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-core:2.60.3
    +-org.apache.ant:ant:1.8.4
and
+-org.jenkins-ci.plugins:leaptest-plugin:3.1.0
  +-org.jenkins-ci.main:jenkins-test-harness:2.13
    +-org.jenkins-ci.lib:lib-jenkins-maven-embedder:3.11
      +-org.apache.ant:ant:1.9.2
```
which gives: `      +-org.apache.ant:ant:1.9.2`

**Comment by romanchetto** [comment]
Hello @bicschneider. thank you for your help. Fixed in 3.1.1
